tests/integration/components/react-component-test.js

```diff
@@ -323,6 +323,36 @@ describeComponent(
 
         expect(this.$().text()).to.contain('Hello Morgan');
       });
+
+      it('supports React-style component file name', function() {
+        this.render(hbs`{{react-component "react-style-file-name"}}`);
```

Comment:
I think we should have a test for `{{react-component "ReactStyleFileName"}} as well